src/Components/Components/src/Routing/RouteTableFactory.cs

```diff
@@ -112,9 +113,11 @@ private static string[] GetTemplates(Type componentType)
     [UnconditionalSuppressMessage("Trimming", "IL2067", Justification = "Application code does not get trimmed, and the framework does not define routable components.")]
     internal static RouteTable Create(Dictionary<Type, string[]> templatesByHandler, IServiceProvider serviceProvider)
     {
+        var routeOptions = Options.Create(new RouteOptions());
+        routeOptions.Value.SetParameterPolicy("regex", typeof(RegexInlineRouteConstraint));
```

Comment:
I'm unsure if this is part of dealing with encoding or if it's addressing a separate issue. Could you clarify?

Reply:
This is addressing a separate issue, we weren't wiring up the regex, which was needed to test this fully, as it doesn't affect other constraints that the Blazor router supports and we don't have support for authoring custom constraints either.

Reply:
Have we effectively added a whole new feature with this line of code then? I'm totally fine with this being added, assuming it doesn't appreciably impact the default bundle size for a Blazor WebAssembly app.

Could you confirm if that's the case?

Reply:
I'm in the process of confirming the bundle size, it's not a feature, it's a bug fix IMO. It works on SSR routing but won't work once it becomes interactive.

Reply:
Thanks, please do confirm! I think we still have a choice of whether to include that feature for interactive mode, given that it's never worked on interactive mode in the past.

Reply:
It turns out that this regressed the size, so I'm going to take it out into its own separate PR that enables this conditionally on WebAssembly behind a feature flag. The feature flag will also need a change in the Blazor SDK to control it, but for the purposes of this repo, we'll keep it enabled.

![image](https://github.com/dotnet/aspnetcore/assets/6995051/03d150a3-6edb-4873-94db-06cdd474a793)